File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/Category.java
package ro.danserboi.quotesformindandsoul;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Data model for each element of the RecyclerView from the MainActivity
 */
class Category implements Parcelable {
    private final int imageResource;
    private final String title;

    public Category(String title, int imageResource) {
        this.title = title;
        this.imageResource = imageResource;
    }

    private Category(Parcel in) {
        title = in.readString();
        imageResource = in.readInt();
    }

    public static final Creator<Category> CREATOR = new Creator<Category>() {
        @Override
        public Category createFromParcel(Parcel in) {
            return new Category(in);
        }

        @Override
        public Category[] newArray(int size) {
            return new Category[size];
        }
    };

    public String getTitle() {
        return title;
    }

    public int getImageResource() {
        return imageResource;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(title);
        dest.writeInt(imageResource);
    }
}

File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/Category.kt
package ro.danserboi.quotesformindandsoul

import android.os.Parcelable
import android.os.Parcel

/**
 * Data model for each element of the RecyclerView from the MainActivity
 */
internal class Category : Parcelable {
    val imageResource: Int
    val title: String?

    constructor(title: String?, imageResource: Int) {
        this.title = title
        this.imageResource = imageResource
    }

    private constructor(`in`: Parcel) {
        title = `in`.readString()
        imageResource = `in`.readInt()
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Category?> = object : Parcelable.Creator<Category?> {
            override fun createFromParcel(`in`: Parcel): Category {
                return Category(`in`)
            }

            override fun newArray(size: Int): Array<Category?> {
                return arrayOfNulls(size)
            }
        }
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(title)
        dest.writeInt(imageResource)
    }
}